test/unit_tests/test_braket_device.py

```diff
@@ -271,6 +297,30 @@ def test_batch_execute_non_parallel(monkeypatch):
         assert res == 1967
 
 
+@patch.object(AwsDevice, "run")
+def test_batch_execute_non_parallel_tracker(mock_run):
```

Comment:
nit: Missing test docstring (which is present on other tests)

Reply:
added.